51/clamytoe/nba.py

```diff
@@ -0,0 +1,178 @@
+DATA_CACHED = 'nba.data'
+NBA_DB = 'nba.db'
+
+# start clean
+if os.path.isfile(NBA_DB):
+    os.remove(NBA_DB)
+
+Player = namedtuple('Player', ('name year first_year team college active '
+                               'games avg_min avg_points'))
+
+conn = sqlite3.connect(NBA_DB)
+cur = conn.cursor()
+
+
+def _get_csv_data():
+    """GIVEN:
+       Load in CSV data in from remote URL or local cache file"""
+    if os.path.isfile(DATA_CACHED):
+        with open(DATA_CACHED) as f:
+            return f.read()
+    else:
+        with requests.Session() as session:
+            return session.get(DATA_URL).content.decode('utf-8')
+
+
+def load_data():
+    """GIVEN:
+       Converts NBA CSV data into a list of Player namedtuples"""
+    content = _get_csv_data()
+    reader = csv.DictReader(content.splitlines(), delimiter=',')
+    for row in reader:
+        player = Player(name=row['Player'],
+                        year=row['Draft_Yr'],
+                        first_year=row['first_year'],
+                        team=row['Team'],
+                        college=row['College'],
+                        active=row['Yrs'],
+                        games=row['Games'],
+                        avg_min=row['Minutes.per.Game'],
+                        avg_points=row['Points.per.Game'])
+        yield player
+
+
+# CODE HERE (tests under __main__):
+
+def import_to_db(players=None):
+    """Create database table in sqlite3 and import the players data
+
+       required table SQL:
+       CREATE TABLE players (name, year, first_year, team, college,
+                             active, games, avg_min, avg_points)
+    """
+    create_players_table = """
+    -- players table
+    CREATE TABLE IF NOT EXISTS players (
+        name text NOT NULL,
+        year int NOT NULL,
+        first_year int NOT_NULL,
+        team text NOT NULL,
+        college text NOT NULL,
+        active int NOT NULL,
+        games int NOT NULL,
+        avg_min float NOT NULL,
+        avg_points float NOT NULL
+    );
+    """
+    cur.execute(create_players_table)
+    if players is None:
+        add_player_sql = f'''
+        INSERT INTO players VALUES (?, ?, ?, ?, ?, ?, ?, ?, ?);'''
+        cur.executemany(add_player_sql, list(load_data()))
+        conn.commit()
+
+
+def _total_records(column):
+    """Helper function to get the total records"""
+    sql = f'''SELECT COUNT({column}) FROM players'''
+    return cur.execute(sql).fetchone()[0]
+
+
+def player_with_max_points_per_game():
+    """The player with highest average points per game (don't forget to CAST to
+       numeric in your SQL query)"""
+    sql = '''SELECT name, MAX(avg_points) FROM players'''
+    return cur.execute(sql).fetchone()[0]
+
+
+def number_of_players_from_duke():
+    """Return the number of players with college == Duke University"""
+    sql = '''SELECT COUNT(*) FROM players WHERE college == "Duke University"'''
```

Comment:
Being picky but in SQL it would be = or LIKE:
https://www.w3schools.com/sql/sql_operators.asp